### ISSUE TITLE
find a font that doesn't have terrible Perso-Arabic support

### ISSUE THREAD

**Issue by jonorthwash** [goavki/apertium-html-tools#17]
The current font is great for Latin, Cyrillic, and Hebrew scripts, but Perso-Arabic script looks terrible.  Mostly it's having trouble with non-Arabic letters and combinations.

![screenshot_2015-08-21_005853](https://cloud.githubusercontent.com/assets/963849/9401606/e99f090e-479f-11e5-85ba-45d3a97bb2e4.png)

**Comment by sushain97** [comment]
I'm definitely missing the buggy part of this. Could you point it out?

**Comment by jonorthwash** [comment]
Yeah, the non-Arabic letters aren't connecting to one another.  E.g., the line at the top should be like this:
![screenshot from 2016-12-18 19-08-32](https://cloud.githubusercontent.com/assets/963849/21297851/899270ca-c555-11e6-84b9-fc403578c237.png)

The next-to-last word analysed should look like this:
![screenshot from 2016-12-18 19-10-51](https://cloud.githubusercontent.com/assets/963849/21297874/d1b53680-c555-11e6-9ef6-c788aa3027bd.png)

Also, the arrows are screwed up.  Was that a separate issue?

**Comment by sushain97** [comment]
From https://sourceforge.net/p/apertium/tickets/27/:

> **Sushain:** Html-tools is built to load the Ubuntu font via Google Fonts with only Latin glyphs. Functionality should be added such that the Cyrillic extension of the font is loaded as well when needed (change in HTML for e.g. index.tat.html and change with JS for selection via dropdown). Similarly, better fonts should be loaded for Arabic script, etc. It would be interesting if browsers didn't load the font until it was actually used in which case this could be significantly simpler...
> > **Jonathan:** Perhaps we should load some defaults for now. That is, good rendering in exchange for slower load times. Currently we have good load time, but poor rendering...

>> **Sushain:** A possible way to address this is using Google's Noto font:
https://code.google.com/p/noto/
http://www.google.com/get/noto/#/
We can provide the interface with a mapping of languages -> font using one of the Noto fonts. For example, if the interface is switched to Hebrew, the site will load Noto Sans Hanunoo. This will help provide a more consistent aesthetic.

Google's approach to this is just using Arial virtually everywhere on the Translate page.

**Comment by jonorthwash** [comment]
Does Arial support enough stuff for our purposes?

**Comment by sushain97** [comment]
@jonorthwash http://stackoverflow.com/a/378196/1266600 font replacement confuses me...

**Comment by sushain97** [comment]
I like: 
![image](https://cloud.githubusercontent.com/assets/4211838/21433486/97cb3392-c835-11e6-94b7-ef3c79c270a3.png)

**Comment by jonorthwash** [comment]
I'm happy to go with Noto.

**Comment by share-with-me** [comment]
Google Noto seem the current go-to fonts. Although I have never used these, preliminary browsing gave a few more suggestions and links. Just posting here in case, they are felt needed.
1. [https://www.google.com/get/noto/](https://www.google.com/get/noto/)
2. [http://stackoverflow.com/questions/7185106/what-are-my-web-font-choices-for-arabic](http://stackoverflow.com/questions/7185106/what-are-my-web-font-choices-for-arabic)

**Comment by jonorthwash** [comment]
Characters are also displaying wrong in Marathi:
![2018-02-24-16 28 45_001](https://user-images.githubusercontent.com/963849/36635253-d66bcdcc-197f-11e8-818d-d8625674cec4.png)

E.g., झ and खा.

**Comment by shardulc** [comment]
Devanagari characters look pretty good in Noto.

**Comment by jonorthwash** [comment]
And it's not good for Tatra either!

Note the different heights of ө, ә, ү, җ as commpared to other characters.  (Though the ә is okay in the right corner of the screenshot.)
![2018-02-27-23 29 20_001](https://user-images.githubusercontent.com/963849/36770119-5b633a9a-1c16-11e8-9c8c-548989cd43c4.png)

**Comment by jonorthwash** [comment]
@sushain97, on IRC you shared reasoning why switching to Noto is harder than it sounds.  Could you explain here too?

**Comment by sushain97** [comment]
Putting this here so I can read it fully later:
https://m.mediawiki.org/wiki/Universal_Language_Selector/WebFonts.

**Comment by sushain97** [comment]
Using Noto is involved because:

1. we need some place to associate language codes => fonts (should it be `locales.json` or another file?)
2. that place needs to be integrated into the build system
3. we need to ensure that
    - on page load: load correct font and adjust corresponding CSS (we need to prevent FOUC unless we embed the correct font at build time (complicated with our outmoded build system))
    - on language selector change, load correct font and adjust corresponding CSS 
4. ensure that the language selectors don't get bugged by the font changes

https://github.com/wikimedia/jquery.webfonts could be useful in making step 3 easier but it also might be optimal to roll our own

That is one of three options. The other two are:
- use the system font stack: https://css-tricks.com/snippets/css/system-font-stack/
- use Arial and let system font replacement handle the rest. this is what Google does. see  http://stackoverflow.com/a/378196/1266600

Regardless, each of these options needs to be tested to make sure it actually solves our problems. IF it has all the scripts we need, Noto will likely lead to the most consistent UX and is least fragile BUT it is more complicated and results in higher bandwidth usage.
